File: chess-app/src/engine/pieces/Bishop.js
import Piece from './Piece.js';

export default class Bishop extends Piece {
    constructor(color) {
        super('bishop', color);
    }
    getPseudoLegalMoves(board, fromPos){
        let directions = [[1,1],[-1,1],[1,-1],[-1,-1]]
        return this._getSlidingMoves(directions, board, fromPos)
    }

    deepCopy() {
        const copy = new Bishop(this.color);
        copy.hasMoved = this.hasMoved;
        return copy;
    }
}

File: chess/src/main/java/com/app/chess/Pieces/Bishop.java
package com.app.chess.Pieces;
import java.util.List;

import com.app.chess.*;

public class Bishop extends Piece {
    public Bishop(Player color) {
        super(PieceType.BISHOP, color);
    }

    public Bishop() {
        super(PieceType.BISHOP);
    }

    @Override
    public List<int []> getPsuedoLegalMoves(Board board, int[] fromPos) {
        List<int[]> directions = List.of(
            new int[]{1,1}, new int[]{1,-1},
            new int[]{-1,1}, new int[]{-1,-1}
        );
        return getSlidingMoves(directions, board, fromPos);
    }

    @Override
    public Piece deepCopy() {
        Bishop copy = new Bishop(this.color);
        copy.hasMoved = this.hasMoved;
        return copy;
    }
}
